File: iosApp/iosApp/HomeScreen.swift
//
//  HomeScreen.swift
//  iosApp
//
//  Created by Julio Luque on 21/11/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import SwiftUI
import shared

struct HomeScreen: View {
    let lastOrders: [Order]
    let stories: [Story]
    let products: [Product]

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                Text("Last orders")
                    .font(.headline)
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack {
                        ForEach(lastOrders, id: \.deliveryDate) { order in
                            VStack {
                                Text("DELIVERY DATE:")
                                    .font(.caption)
                                Text(order.deliveryDate)
                                    .font(.body)
                                Button(order.status) {
                                    // Acción
                                }
                                .buttonStyle(.bordered)
                            }
                            .frame(width: 150, height: 80)
                            .background(Color.gray.opacity(0.2))
                            .cornerRadius(8)
                        }
                    }
                }

                Text("Stories about producers")
                    .font(.headline)
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack {
                        ForEach(stories, id: \.title) { story in
                            VStack {
                                Text(story.title)
                                    .font(.body)
                                Text(story.description)
                                    .font(.caption)
                            }
                            .frame(width: 300, height: 200)
                            .background(Color.gray.opacity(0.2))
                            .cornerRadius(8)
                        }
                    }
                }

                Text("New products")
                    .font(.headline)
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack {
                        ForEach(products, id: \.name) { product in
                            VStack {
                                Text(product.name)
                                    .font(.body)
                                Text(product.description)
                                    .font(.caption)
                            }
                            .frame(width: 150, height: 200)
                            .background(Color.gray.opacity(0.2))
                            .cornerRadius(8)
                        }
                    }
                }
            }
            .padding()
        }
    }
}

File: composeApp/src/main/java/com/jetbrains/mula/ui/screens/HomeScreen.kt
package com.jetbrains.mula.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.jetbrains.screens.models.Order
import com.jetbrains.screens.models.Story
import com.jetbrains.screens.models.Product

@Composable
fun HomeScreen(
    lastOrders: List<Order>,
    stories: List<Story>,
    products: List<Product>
) {
    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        // Sección "Last Orders"
        Text("Last orders", style = MaterialTheme.typography.titleLarge)
        LazyRow {
            items(lastOrders.size) { index ->
                val order = lastOrders[index]
                Card(
                    modifier = Modifier
                        .padding(8.dp)
                        .width(150.dp)
                        .height(80.dp),
                    elevation = CardDefaults.cardElevation(4.dp) // Corregido
                ) {
                    Column(
                        modifier = Modifier.padding(8.dp)
                    ) {
                        Text("DELIVERY DATE:", style = MaterialTheme.typography.labelSmall)
                        Text(order.deliveryDate.toString(), style = MaterialTheme.typography.bodyMedium)
                        Button(onClick = { /* Acción de reordenar */ }) {
                            Text(order.status)
                        }
                    }
                }
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        // Sección "Stories"
        Text("Stories about producers", style = MaterialTheme.typography.titleLarge)
        LazyRow {
            items(stories.size) { index ->
                val story = stories[index]
                Card(
                    modifier = Modifier
                        .padding(8.dp)
                        .width(300.dp)
                        .height(200.dp),
                    elevation = CardDefaults.cardElevation(4.dp) // Corregido
                ) {
                    Column(modifier = Modifier.padding(8.dp)) {
                        Text(story.title, style = MaterialTheme.typography.bodyLarge)
                        Text(story.description, style = MaterialTheme.typography.bodyMedium)
                    }
                }
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        // Sección "New Products"
        Text("New products", style = MaterialTheme.typography.titleLarge)
        LazyRow {
            items(products.size) { index ->
                val product = products[index]
                Card(
                    modifier = Modifier
                        .padding(8.dp)
                        .width(150.dp)
                        .height(200.dp),
                    elevation = CardDefaults.cardElevation(4.dp) // Corregido
                ) {
                    Column(modifier = Modifier.padding(8.dp)) {
                        Text(product.name, style = MaterialTheme.typography.bodyLarge)
                        Text(product.description, style = MaterialTheme.typography.bodyMedium)
                    }
                }
            }
        }
    }
}
